test/sharness/t0063-daemon-init.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+#
+# Copyright (c) 2014 Juan Batiz-Benet
+# MIT Licensed; see the LICENSE file in this repository.
+#
+
+test_description="Test daemon --init command"
+
+. lib/test-lib.sh
+
+# We don't want the normal test_init_ipfs but we need to make sure the
+# IPFS_PATH is set correctly.
+export IPFS_PATH="$(pwd)/.ipfs"
+
+# safety check since we will be removing the directory
+if [ -e "$IPFS_PATH" ]; then
+	echo "$IPFS_PATH exists"
+	exit 1
+fi
+
+test_ipfs_daemon_init() {
+	# Doing it manually since we want to launch the daemon with an
+	# empty or non-existent repo; the normal
+	# test_launch_ipfs_daemon does not work since it assumes the
+	# repo was created a particular way with regard to the API
+	# server.
+
+	test_expect_success "'ipfs daemon --init' succeeds" '
+		ipfs daemon --init >actual_daemon 2>daemon_err &
```

Comment:
(lets not skip the test, I think its okay if it fails and says it needs those ports)